### DISCUSSION TITLE
Remove black formatting/linting

### DISCUSSION THREAD

**Discussion by zain-sohail** [OpenCOMPES/sed#126] [Ideas]
As mentioned by @steinnymir in the PR #118, the black formatting is sometimes very unpleasant and just explodes the lines in the code. Though, it has advantages in readability sometimes, many times it doesn't. 

I'd propose we remove it from the pre commit hooks and just stick to linter definitions given in the github workflow. 

Any suggestions are welcome @rettigl @daviddoji regarding linting and code formatting.

**Comment by daviddoji** [comment]
I don't think it's a good idea to get rid of an auto-formatter such as black. Passing black as a pre-commit hook will take the burden from the developer to pay attention to tiny aesthetics details and, more importantly, the code will always be formatted in the same way.

I wouldn't remove it if I just find a line or two (from thousands) "ugly", which it is basically a personal opinion.

Having a standard is more important than the standard being excellent.

**Comment by rettigl** [comment]
I would also certainly not remove the auto formatter from the pre commit hooks. To improve performance, we can tweak the options somewhat (e. G. Slightly larger line length to match pylint, etc.). There are indeed some cases where edits made by black are marked as errors by pylint (import order in test binning is an example).
The way I use the pre commit hooks is that I run them manually and accept the changes I like, and discard the ones I don't. They will off course come back if you work on the same file.
Regarding readibikity of  code, it could also help just to create intermediate variables, that e. G. Contain the result of the first part of your chain, and then apply the rest to this variable...

**Reply by steinnymir** [reply]
> Regarding readibikity of code, it could also help just to create intermediate variables, that e. G. Contain the result of the first part of your chain, and then apply the rest to this variable...

I think this is the most readable solution.

But I would also argue line lenght increased to 100 would be really nice 😄

**Reply by rettigl** [reply]
Added in PR #132
